File: Homeworks/Homework1/Tire.Kutay.php
<?php

// Function declaration of foo

function foo($key, $value){
    
        echo $key . ": " . $value. "<br>";
}

// Initialization of Associative Array

$phpDictionary = array("Assyria" => "Nineveh", "Babylonia" => "Babylon", "Sumeria" => "Ur", "Persia" => "Susa");

// Getting a value

echo "Getting the value of Sumeria: <br>";
$value1 = $phpDictionary["Sumeria"];
echo "The value of 'Sumeria' is " . $value1 . "<br>";

// Adding a new element

echo "<br>Adding the new element 'Egypt': <br>";
$phpDictionary['Egypt'] = "Thebes";
foreach ($phpDictionary as $key => $value) {
	foo($key, $value);
}

// Removing an element

echo "<br> Removing the element 'Persia': <br>";
unset ($phpDictionary['Persia']);
foreach ($phpDictionary as $key => $value) {
	foo($key, $value);
}

// Modifying an element

echo "<br>Modifying the element 'Assyria': <br>";
$phpDictionary['Assyria'] = "Assur";
foreach ($phpDictionary as $key => $value) {
	foo($key, $value);
}

// Searching for the existence of a key

echo "<br>Searching for 'Babylonia' key: <br>";

if (array_key_exists('Babylonia', $phpDictionary))
    echo "'Babylonia' key is successfully found.<br>";
else
    echo "The key was not found.<br>";

// The key is not found

echo "<br>Searching for 'Rome' key: <br>";

if (array_key_exists('Rome', $phpDictionary))
    echo "'Rome' key is successfully found.<br>";
else
    echo "The key was not found.<br>";

// Searching for the existence of a value

echo "<br>Searching for 'Thebes' value: <br>";

if (array_search('Thebes', $phpDictionary))
    echo "'Thebes' value is successfully found.<br>";
else
    echo "The value was not found.<br>";

// The value is not found

echo "<br>Searching for 'Athens' value: <br>";

if (array_search('Athens', $phpDictionary))
    echo "'Athens' value is successfully found.<br>";
else
    echo "The value was not found.<br>";

// Printing key-value pair

echo "<br> Looping through the array: <br>";
foreach ($phpDictionary as $key => $value) {
	foo($key, $value);
}

?>

File: Homeworks/Homework1/Tire_Kutay.dart
void main() {
  
  // Associative Array Initialization
  
  var dictionaryOfDart = {
    
    'Assyria' : 'Nineveh',
    'Babylonia' : 'Babylon',
    'Sumeria' : 'Ur',
    'Persia' : 'Susa'
  };
  
  // Getting a value
  
  print("Getting the value of 'Sumeria'");
  var getValue = dictionaryOfDart['Sumeria'];
  print("Value of 'Sumeria' is: '$getValue'");
  
  // Adding an element
  
  print('\n');
  print("Adding a new element 'Egypt':");
  dictionaryOfDart.addAll({'Egypt' : 'Thebes'});
  print(dictionaryOfDart);
  print('\n');
  
  // Removing an element
  
  print("Removing the element 'Persia':");
  var removedElement = dictionaryOfDart.remove('Persia');
  
  if (removedElement != null) {
    
    print("'Persia' is successfully removed.");
  }
  else {
    
    print("'Persia' does not exist.");
  }
  print(dictionaryOfDart);
  print('\n');
  
  // Modifying an element
  
  print("Modifying the element 'Assyria': ");
  dictionaryOfDart['Assyria'] = 'Assur';
  print(dictionaryOfDart);
  print('\n');
  
  // Searching for the existence of a key
  
  print("Searching for the 'Babylonia' key:");
  
  if(dictionaryOfDart.containsKey('Babylonia')){
    
    var matchingValue = dictionaryOfDart['Babylonia'];
    print("Key is successfully found with the value '$matchingValue'.\n");
    
  }
  else{
    
    print('Key was not found\n');
  }
  
  print("Searching for the 'Rome' key:");
  if(dictionaryOfDart.containsKey('Rome')){
    
    var matchingValue = dictionaryOfDart['Rome'];
    print("Key is successfully found with the value '$matchingValue'.\n");
    
  }
  else{
    
    print('Key was not found.\n');
  }
  
  // Searching for the existence of a value
  
  print("Searching for the 'Thebes' value:");
  
  if(dictionaryOfDart.containsValue('Thebes')){
    

    print("'Thebes' is successfully found.\n");
    
  }
  else{
    
    print("'Thebes' was not found as a value.\n");
  }
  
  print("Searching for the 'Athens' value:");
  if(dictionaryOfDart.containsValue('Athens')){
    
    print("'Athens' is successfully found.\n");
    
  }
  else{
    
    print("'Athens' was not found as a value.\n");
  }
  
 
  // Function declaration and printing key-value pair
  
  print("Looping through the array: ");
  void foo(String key, String value) {

    print('$key: $value');

  }
    
  dictionaryOfDart.forEach((key,value) {

    foo(key,value);

  });
  
}
